cmd/kubelet/app/server.go

```diff
@@ -515,11 +515,16 @@ func InitializeTLS(kc *componentconfig.KubeletConfiguration) (*server.TLSOptions
 		kc.TLSCertFile = path.Join(kc.CertDirectory, "kubelet.crt")
 		kc.TLSPrivateKeyFile = path.Join(kc.CertDirectory, "kubelet.key")
 		if !certutil.CanReadCertOrKey(kc.TLSCertFile, kc.TLSPrivateKeyFile) {
-			cert, key, err := certutil.GenerateSelfSignedCertKey(nodeutil.GetHostname(kc.HostnameOverride), nil, nil)
+			caCert, _, cert, key, err := certutil.GenerateSelfSignedCertKey(nodeutil.GetHostname(kc.HostnameOverride), nil, nil)
 			if err != nil {
 				return nil, fmt.Errorf("unable to generate self signed cert: %v", err)
 			}
 
+			caCertPath := path.Join(path.Dir(kc.TLSCertFile), "ca-"+path.Base(kc.TLSCertFile))
+			if err := certutil.WriteCert(caCertPath, caCert); err != nil {
+				return nil, err
+			}
+
 			if err := certutil.WriteCert(kc.TLSCertFile, cert); err != nil {
```

Comment:
Yes, I like that. This also solves the logical issue below when to create a new root ca.